controllers/core/openstackcontrolplane_controller.go

```diff
@@ -418,6 +441,23 @@ func (r *OpenStackControlPlaneReconciler) reconcileNormal(ctx context.Context, i
 	return ctrl.Result{}, nil
 }
 
+func (r *OpenStackControlPlaneReconciler) reconcileDelete(ctx context.Context, instance *corev1beta1.OpenStackControlPlane, version *corev1beta1.OpenStackVersion, helper *common_helper.Helper, versionHelper *common_helper.Helper) (ctrl.Result, error) {
+	helper.GetLogger().Info("reconcile delete")
+	controllerutil.RemoveFinalizer(version, versionHelper.GetFinalizer())
```

Comment:
This will return `true` if it actually removes the finalizer, so we could gate on that and only call `versionHelper.PatchInstance` if needed

Reply:
ack, good call. I will add try to add this.

Reply:
ohh sorry I mixed up. In this case we are removing the finalizer from the version object during control plane delete. It make sense to make the version object update conditional. What confused me that we have helper for both the instance (controlplane) and the version object. So here I'm OK to add something like:
```golang
if controllerutil.RemoveFinalizer(version, versionHelper.GetFinalizer()){
   err := versionHelper.PatchInstance(ctx, version)
   if err != nil {
       return ctrl.Result{}, err
    }
}
```